src/onegov/activity/iso20022.py

```diff
@@ -38,11 +38,11 @@ def __repr__(self):
 
     @cached_property
     def references(self):
+        if self.reference:
+            return {self.reference}
         return set(self.extract_references())
```

Comment:
I was thinking that once `self.reference` is present, there's no need to go any further. We can just return it. (assuming we have a reference schema).